plugins/obs-webrtc/whip-output.cpp

```diff
@@ -134,24 +140,70 @@ void WHIPOutput::ConfigureVideoTrack(std::string media_stream_id,
 		media_stream_track_id.c_str(),
 	};
 
+	// generate the random starting ts for the video track
+	uint32_t rtp_video_timestamp = generate_random_u32();
+
 	rtcPacketizationHandlerInit packetizer_init = {
 		video_ssrc,
 		cname.c_str(),
 		video_payload_type,
 		video_clockrate,
 		0,
-		0,
+		rtp_video_timestamp,
 		RTC_NAL_SEPARATOR_START_SEQUENCE,
-		0};
+		max_fragment_size};
 
 	video_track = rtcAddTrackEx(peer_connection, &track_init);
 	rtcSetH264PacketizationHandler(video_track, &packetizer_init);
 	rtcChainRtcpSrReporter(video_track);
 	rtcChainRtcpNackResponder(video_track, 1000);
 }
 
-bool WHIPOutput::Setup()
+/**
+ * @brief Init before OPTIONS and Setup due to the need of the endpoint url for SendOptions.
+ * 
+ */
+bool WHIPOutput::Init()
 {
+	// construct operating system info string
+	std::stringstream os_info;
+#ifdef _WIN32
+	os_info << "Windows";
+#elif _WIN64
+	os_info << "Windows x86_64";
+#elif __APPLE__
+	os_info << "MacOs";
+#elif __OpenBSD__
+	os_info << "OpenBSD";
+#elif __FreeBSD__
+	os_info << "FreeBSD ";
+#elif __linux__ && __LP64__
+	os_info << "Linux x86_64";
+#else
+	os_info << "Linux";
+#endif
+	// construct information for our user-agent
+	uint32_t obs_ver = obs_get_version();
+	// build the user-agent string
+	std::stringstream ua;
+	// user agent header prefix
+	ua << "User-Agent: Mozilla/5.0 ";
+	// obs version info
+	ua << "(OBS-Studio/" << std::to_string(obs_ver >> 24) << "."
+	   << std::to_string((obs_ver >> 16) & 0xFF) << "."
+	   << std::to_string(obs_ver & 0xFFFF) << "; ";
```

Comment:
Maybe `obs_get_version_string()` is ok.
```suggestion
	ua << "(OBS-Studio/" << obs_get_version_string() << "; ";
```